tasks.py

```diff
@@ -37,3 +44,21 @@ def collect_over_thirty(self) -> List[Dict]:
                 self.over_thirty_records.append(record)
 
         return self.over_thirty_records
+
+
+class CompanyFieldGenerator:
+    def __init__(self, user_records, company_records) -> None:
+        self.user_records = user_records
+        self.company_records = company_records
+
+    def company_id_to_field(self) -> List[Dict]:
+        for record in self.user_records:
```

Comment:
The complexity of these loops is O(nˆ2)

If you give it a second thought you could make it linear (hint: using a dict)